### ISSUE TITLE
No login screen

### ISSUE THREAD

**Issue by CTLS** [estevez-dev/ha_client#357]
When trying to login on latest version.  When I clear cache sometimes I get a nginx error 403.  Sometimes it just shows the empty square symbol instead.

Logs below

0329 20:54:25 [Debug] :  init...
0329 20:54:26 [Debug] :  AppLifecycleState.inactive
0329 20:54:26 [Debug] :  AppLifecycleState.resumed
0329 20:54:26 [Warning] :  Previous connection pending...
0329 20:54:26 [Debug] :  Launching OAuth...
0329 20:54:39 [Warning] :  Previous connection pending...
0329 20:54:50 [Debug] :  Settings changed. Saving...
0329 20:54:50 [Debug] :  Settings change event: reconnect=true
0329 20:54:50 [Warning] :  Previous connection pending...
0329 20:55:26 [Debug] :  Settings changed. Saving...
0329 20:55:26 [Debug] :  Settings change event: reconnect=true
0329 20:55:26 [Warning] :  Previous connection pending...
0329 20:55:37 [Debug] :  Settings changed. Saving...
0329 20:55:37 [Debug] :  Settings change event: reconnect=true
0329 20:55:37 [Warning] :  Previous connection pending...
0329 20:55:55 [Debug] :  Settings changed. Saving...
0329 20:55:55 [Debug] :  Settings change event: reconnect=true
0329 20:55:55 [Warning] :  Previous connection pending...
0329 20:56:02 [Warning] :  Previous connection pending...
0329 20:56:37 [Debug] :  Settings changed. Saving...
0329 20:56:37 [Debug] :  Settings change event: reconnect=true
0329 20:56:37 [Warning] :  Previous connection pending...
0329 20:56:39 [Debug] :  AppLifecycleState.inactive
0329 20:56:39 [Debug] :  AppLifecycleState.paused
0329 20:56:53 [Debug] :  AppLifecycleState.inactive
0329 20:56:53 [Debug] :  AppLifecycleState.resumed
0329 20:56:53 [Warning] :  Previous connection pending...
0329 20:56:54 [Debug] :  AppLifecycleState.inactive
0329 20:56:54 [Debug] :  AppLifecycleState.paused
0329 22:02:21 [Debug] :  AppLifecycleState.inactive
0329 22:02:21 [Debug] :  AppLifecycleState.resumed
0329 22:02:21 [Warning] :  Previous connection pending...
0329 22:02:28 [Debug] :  Settings was not changed
0329 22:02:36 [Debug] :  AppLifecycleState.inactive
0329 22:02:38 [Debug] :  AppLifecycleState.paused
0329 22:03:04 [Debug] :  AppLifecycleState.inactive
0329 22:03:04 [Debug] :  AppLifecycleState.resumed
0329 22:03:04 [Warning] :  Previous connection pending...

**Comment by klejejs** [comment]
Here is my nginx config and everything is working perfectly:

```
server {
	listen              443 ssl;
	server_name         example.com
	ssl_certificate     your_cert.cert;
	ssl_certificate_key your_cert.key;
	ssl_protocols       TLSv1 TLSv1.1 TLSv1.2;
	ssl_ciphers         HIGH:!aNULL:!MD5;
	location / {
		proxy_read_timeout 10080m;
		proxy_set_header Host $host;
		proxy_http_version 1.1;
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		proxy_set_header Upgrade $http_upgrade;
		proxy_set_header Connection "upgrade";
		proxy_buffering               off;
		proxy_ssl_verify              off;
		proxy_pass http://192.168.88.250:8123/;
	}
	location /api/websocket {
		proxy_read_timeout 10080m;
		proxy_set_header Host $host;
		proxy_http_version 1.1;
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		proxy_set_header Upgrade $http_upgrade;
		proxy_set_header Connection "upgrade";
		proxy_buffering               off;
		proxy_ssl_verify              off;
		proxy_pass http://192.168.88.250:8123/api/websocket;
	}
}
```

I use https and in app setup I set the port to be 443.

**Comment by CTLS** [comment]
I just recreated the proxy setup with nginx proxy manager and it fixed the issue so I think this was more of a reverse proxy issue than anything else.  I did notice I had to turn off "block common exploits" in order to access which is concerning as that was never an issue with the web page but it is with the app.  Is there a way to see the web link it is currently using to login?  I want to try the same exact webpage through browser and see if it still has issue with the old nginx settings.

**Comment by estevez-dev** [comment]
They're is not only a link. There is also a POST request to get access token. May be the issue was there. Toy can find all needed information on authentication in [Home Assistant docs](https://developers.home-assistant.io/docs/en/auth_api.html).
Any way I will leave this issue open because this error was not handled correctly.

**Comment by estevez-dev** [comment]
Fixed with [b277363](https://github.com/estevez-dev/ha_client/commit/b2773635f5d284d45267287af1184c8a443830ae)
